lib/formatters/json.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * @fileoverview JSON reporter
+ * @author Burak Yigit Kaya aka BYK
+ * @copyright 2015 Burak Yigit Kaya. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results) {
+    var output = {};
+
+    results.forEach(function(result) {
+        output[result.filePath] = (output[result.filePath] || []).concat(result.messages);
+    });
+
+    return JSON.stringify(output);
```

Comment:
We shouldn't output a JSON format that has no relationship to what ESLint outputs. So we should either stringify what is passed in without modification or mimic what ESLint produces: http://eslint.org/docs/developer-guide/nodejs-api#executeonfiles